packages/rum-core/src/domain/tracing/tracer.ts

```diff
@@ -121,7 +137,9 @@ function injectHeadersIfTracingAllowed(
   context.traceId = new TraceIdentifier()
   context.spanId = new TraceIdentifier()
   context.traceSampled = !isNumber(configuration.traceSampleRate) || performDraw(configuration.traceSampleRate)
-  inject(makeTracingHeaders(context.traceId, context.spanId, context.traceSampled, tracingOption.propagatorTypes))
+  if (configuration.traceContextInjection === TraceContextInjection.All) {
+    inject(makeTracingHeaders(context.traceId, context.spanId, context.traceSampled, tracingOption.propagatorTypes))
+  }
```

Comment:
FMU, the check should more look like this 

```ts
 if (
    (context.traceSampled && configuration.traceContextInjection === TraceContextInjection.Sampled) ||
    configuration.traceContextInjection === TraceContextInjection.All
  ) {
    inject(makeTracingHeaders(context.traceId, context.spanId, context.traceSampled, tracingOption.propagatorTypes))
  }
```

Reply:
Thank you! I updated this and added tests accordingly.